articles/variables-in-js/index.md

```diff
@@ -0,0 +1,132 @@
+
+When a variable is block-scoped, it means that it exists only inside the block where it was defined.
+
+### Naming Variables In JavaScript
+Before you start declaring variables, you should learn how to name them. Below are some of the rules that you should follow. Failing to follow these rules will cause an error.
+
+- Variables should not have spaces.
+- Variables should begin with an underscore, a dollar sign, or a letter.
+- Variables are very case-sensitive. This means `car_TYPE` is treated as an entirely different variable than one named `car_type`
+- Variables must contain only letters, numbers, underscores, or dollar signs.
+- You can't use any JavaScript *reserved words* as a variable name. Reserved words include; private, public, enum, etc.
+
+### Declaring variables using the keyword `var`
+Variables can be declared using the `var` keyword as shown below:
+
+```javascript
+var favouriteThing;
+```
+
+Upon declaring a variable, you can assign a value to the variable such as a string or a number.
+
+Example:
+
+```javascript
+var favouriteThing;
+favouriteThing = "Icecream!";
+```
+
+Declaring and initializing a variable at the same time is also possible using the following syntax.
+```javascript
+var favouriteThing = "Icecream!";
+```
+
+Declaring two or more variables in a single statement is also possible.
+Each declaration is separated by a comma (`,`).
+
+Example:
+
+```javascript
+var favouriteThing = "Icecream!", bestMovie = "Code Eight";
+```
+
+The `var` keyword is used to declare a function-scoped variable and a globally-scoped variable. 
+
+If you use `var` inside a block, the variable will not be block scoped. It will either be function scoped or globally scoped depending upon where the block is present. 
+
+Example:
+
+```javascript
+var x = 2;
+
+if (x === 2) {
+  var x = 3;
+  console.log(x); // expected output: 3
+}
+
+console.log(x); //expected output: 3
+```
+
+### Declaring variables using the keyword `let`
+The main difference between keywords `var` and `let` is that variables declared using `let` are block-scoped, while `var` is function scoped.
+
+Example:
+```javascript
+var a = 20, b = 10;
+{
+  let exe = a;
+  a = b;
+  b = exe;
+}
+console.log(exe); // Uncaught ReferenceError: exe is not defined
+```
+In the above example, the `exe` variable only exists inside the block, and referencing it outside the block will cause a reference error.
+
+### Declaring variables using the keyword  `const`
+The difference between the `const` and other keywords used to declare variables is that a variable declared using the keyword `const` must be initialized immediately with a value, and that value **can't be redeclared**. 
+
+`const` keyword is used to declare block scope variables.
+
+Example:
+
+```javascript
+const pi = 3.14;
+pi = 1.5; //SyntaxError: Identifier 'pi' has already been declared
+```
+
+The error: `SyntaxError: Identifier 'pi' has already been declared`, informs you that you can't change the value of `pi`. 
```

Comment:
The error you've mentioned is wrong.

The actual error is `Uncaught TypeError: Assignment to constant variable.`